Version Generation.adoc

```diff
@@ -24,19 +24,31 @@ This design covers 2 main changes to WPILib's build system: version number code
 
 ==== WPILibJ and WPILibC
 
-The first two opportunities are basically the same generation, the WPILibJ and WPILibC version numbers. Currently, these are hardcoded at RobotBase.java:218 and RobotBase.cpp:42, with simulation having no knowledge of what version it is at all. I propose having 2 auto-generated files, one for each project. For WPILibC, this file would be a generated header file, generated to wpilibc\shared\include\WPILibVersion.h. We would add this file to the .gitignore to make sure that it is not checked into the project. The file would look something like this:
+The first two opportunities are basically the same generation, the WPILibJ and WPILibC version numbers. Currently, these are hardcoded at RobotBase.java:218 and RobotBase.cpp:42, with simulation having no knowledge of what version it is at all. I propose having auto-generated files that contain the version. For WPILibC, this file would be a generated header file and matching cpp file, generated to wpilibc\shared\include\WPILibVersion.(h|cpp). We would add this file to the .gitignore to make sure that it is not checked into the project. The file would look something like this:
 
+[[WPILibVersion.h]]
 [source,c++]
 ----
 /* BSD License Header */
 #pragma once
 
 /*
- * Autogenerated file! Do not manually edit this file. To change the version number,
- * change the WPILib version file located at <allwpilib_root>/wpilib_version.json
+ * Autogenerated file! Do not manually edit this file. This version is regenerated
+ * any time the publish task is run, or when this file is deleted.
```

Comment:
While true, I guess I just find the idea of having half of this checked in while the other half is generated weird.